drivers/cisco/ise/network_access.cr

```diff
@@ -56,64 +59,30 @@ class Cisco::Ise::Guests < PlaceOS::Driver
     logger.debug { "Basic auth details: #{@basic_auth}" } if @debug
   end
 
-  def create_internal(
-    event_start : Int64,
-    attendee_email : String,
-    attendee_name : String,
-    company_name : String? = nil,         # Mandatory but driver will extract from email if not passed
-    phone_number : String = "0123456789", # Mandatory, use a fake value as default
-    sms_service_provider : String? = nil, # Use this param to override the setting
-    guest_type : String? = nil,           # Mandatory but use this param to override the setting
-    portal_id : String? = nil             # Mandatory but use this param to override the setting
+  def create_internal_user(
+    email : String,
+    name : String? = nil,
+    first_name : String? = nil,
+    last_name : String? = nil,
+    description : String? = nil,
+    password : String? = nil
   )
-    # Determine the name of the attendee for ISE
-    guest_names = attendee_name.split
-    first_name_index_end = guest_names.size > 1 ? -2 : -1
-    first_name = guest_names[0..first_name_index_end].join(' ')
-    last_name = guest_names[-1]
-    username = genererate_username(first_name, last_name)
-    password = genererate_password(first_name, last_name)
-
-    return {"username" => username, "password" => UUID.random.to_s[0..3]}.merge(@custom_data) if setting?(Bool, :test)
-
-    sms_service_provider ||= @sms_service_provider
-    guest_type ||= @guest_type
-    portal_id ||= @portal_id
-
-    time_object = Time.unix(event_start).in(@timezone)
-    from_date = time_object.at_beginning_of_day.to_s(TIME_FORMAT)
-    to_date = time_object.at_end_of_day.to_s(TIME_FORMAT)
-
-    # If company_name isn't passed
-    # Hackily grab a company name from the attendee's email (we may be able to grab this from the signal if possible)
-    company_name ||= attendee_email.split('@')[1].split('.')[0].capitalize
-
-    internal_user = Models::InternalUser.from_json(%({}))
-
-    # These custom attributes and any custom attribute needs to be predefined
-    # in the ISE GUI.
-    # custom_attributes = {
-    #   "fromDate"           => from_date,
-    #   "toDate"             => to_date,
-    #   "location"           => @location.to_s,
-    #   "companyName"        => company_name,
-    #   "phoneNumber"        => phone_number,
-    #   "smsServiceProvider" => sms_service_provider.to_s,
-    #   "guestType"          => guest_type,
-    #   "portalId"           => portal_id,
-    # } of String => String
-
-    # custom_attributes.merge!(@custom_data)
-
-    internal_user.name = username
+    name ||= email
+    password ||= generate_password
+
+    # Not sure if this response is still accurate for ISEv3, todo: check it and align with ISEv3's response
+    # return {"username" => username, "password" => generate_password}.merge(@custom_data) if @test_mode
+
+    # Is there a one line way of doing this?
```

Comment:
You can extend the `{name: name}` NamedTuple and provide the required JSON keys instead of assigning them one by one.